torch/fx/experimental/symbolic_shapes.py

```diff
@@ -242,7 +202,7 @@ def eval(cls, base, divisor):
     'mod': lambda a, b: a % b,
     'pow': lambda a, b: a ** b,
     'truediv': lambda a, b: a / b,
-    'floordiv': lambda a, b: FloorDiv(a, b),
+    'floordiv': lambda a, b: a // b,
```

Comment:
@Chillee 
Current floordiv leads to pretty nasty error. So I am proposing to remove it. WDYT
```python
======================================================================
ERROR: test_method_fn_floordiv_first_type_int_second_type_float (__main__.TestSymNumberMagicMethods)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/albandes/local/pytorch/3.8_debug_source/torch/testing/_internal/common_utils.py", line 248, in instantiated_test
    test(self, **param_kwargs)
  File "test_dynamic_shapes.py", line 538, in test_method
    self._do_test(fn, inp1, inp2, shape_env, is_unary_fn)
  File "test_dynamic_shapes.py", line 498, in _do_test
    self.assertEqual(guard_fn(out), ref_out)
  File "test_dynamic_shapes.py", line 479, in guard_fn
    raise e
  File "test_dynamic_shapes.py", line 474, in guard_fn
    return getattr(v.node, f"guard_{tp}")("", 0)
  File "/home/albandes/local/pytorch/3.8_debug_source/torch/fx/experimental/symbolic_shapes.py", line 188, in guard_int
    return int(self.shape_env.evaluate_expr(self.expr))
  File "/home/albandes/local/pytorch/3.8_debug_source_env/lib/python3.8/site-packages/sympy/core/expr.py", line 321, in __int__
    r = self.round(2)
  File "/home/albandes/local/pytorch/3.8_debug_source_env/lib/python3.8/site-packages/sympy/core/expr.py", line 3817, in round
    raise TypeError(
TypeError: Expected a number but got FloorDiv:
```

Also current impl doesn't seem to follow best practices as it does fusion which is explicitly mentioned as a bad idea: https://docs.sympy.org/latest/guides/custom-functions.html#best-practices-for-eval

Reply:
🤔 This error looks like somehow, `evaluate_expr` is returning a sympy expression and not an integer.

Reply:
This evaluate expr business, where we ask sympy to evaluate int/float ops for us, is awfully questionable. We should just compute it ourselves

Reply:
I don't see why it should be questionable 🤔

Reply:
It's just a complicated way of doing something we can do very simply

Reply:
I don't see how we do it simpler? `evaluate_expr` (other than adding a guards) simply substitutes in all of the symbols.

Reply:
If we only have backed symints, we can just maintain the true value for every intermediate compute. Even with unbacked symints we could maintain the unsimplified expression and eval it like a normal interpreter